kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -7015,11 +7015,42 @@ protected void handleAccessResource(HttpServletRequest req, HttpServletResponse
 					return; 
 				}
 
+				// If there is a direct link to the asset, no sense streaming it.
+				// Send the asset directly to the load-balancer or to the client
+				URI directLinkUri = m_storage.getDirectLink(resource);
```

Comment:
Basic repo steps:

Create an HTML file in resources and paste 10 paragraphs or so of lipsum text ...... Then try to view file ..... experiment with enabling/disabling wrapping the content via Actions -> Edit Details